porter/v1/cluster_control_plane.proto

```diff
@@ -7,15 +7,35 @@ import "porter/v1/aws_assume_role.proto";
 import "porter/v1/cloud_provider_credentials.proto";
 import "porter/v1/contract.proto";
 import "porter/v1/porter_app.proto";
+import "porter/v1/cluster.proto";
+import "porter/v1/errors.proto";
+
+message PreflightCheckRequest {
+  int64 project_id = 1;
+  // cloud_provider [REQUIRED] represents the provider that we will provisioning the cluster in
+  EnumCloudProvider cloud_provider = 2;
+  // cloud_provider_credentials_id [REQUIRED] is the Porter credentials that will be used for provisioning a cluster.
+  // These must be stored within Porter, prior to cluster creation. For AWS this refers to the last link in an assume role chain
+  string cloud_provider_credentials_id = 3;
+}
+
+message PreflightCheckResponse {
+
+  Error error = 1 ; 
```

Comment:
Make this `repeated Error error = 1`. This will allow us to show all of the errors that happen in preflight at once to the user